spec/factories/virtual_template.rb

```diff
@@ -0,0 +1,16 @@
+FactoryGirl.define do
+  factory :virtual_template, :class => 'ManageIQ::Providers::CloudManager::VirtualTemplate' do
+    vendor 'amazon'
+    name 'vt'
+    location 'us-west-1'
+    ems_ref 'i-12345'
+    availability_zone_id 0
+    cloud_subnet_id 1
+    cloud_network_id 2
```

Comment:
@Fryguy sure, will do. only reason for hardcoded values here was because again there's been no implementation of validation on the real values (working on it). but I see your point